daemon/containerd/image_manifest.go

```diff
@@ -78,21 +124,40 @@ func (im *ImageManifest) Metadata() containerdimages.Image {
 	return md
 }
 
-// IsPseudoImage returns false if the manifest has no layers or any of its layers is a known image layer.
-// Some manifests use the image media type for compatibility, even if they are not a real image.
-func (im *ImageManifest) IsPseudoImage(ctx context.Context) (bool, error) {
-	desc := im.Target()
-
+func (im *ImageManifest) IsAttestation() bool {
 	// Quick check for buildkit attestation manifests
 	// https://github.com/moby/buildkit/blob/v0.11.4/docs/attestations/attestation-storage.md
 	// This would have also been caught by the layer check below, but it requires
 	// an additional content read and deserialization of Manifest.
-	if _, has := desc.Annotations[attestation.DockerAnnotationReferenceType]; has {
+	if _, has := im.Target().Annotations[attestation.DockerAnnotationReferenceType]; has {
+		return true
+	}
+	return false
+}
+
+// IsPseudoImage returns false when any of the below is true:
+// - The manifest has no layers
+// - None of its layers is a known image layer.
+// - The manifest has unknown/unknown platform.
+//
+// Some manifests use the image media type for compatibility, even if they are not a real image.
+func (im *ImageManifest) IsPseudoImage(ctx context.Context) (bool, error) {
+	if im.IsAttestation() {
 		return true, nil
 	}
 
+	plat := im.Target().Platform
+	if plat != nil {
+		if plat.OS == "unknown" && plat.Architecture == "unknown" {
+			return true, nil
+		}
+	}
+
 	mfst, err := im.Manifest(ctx)
```

Comment:
I noticed that `Manifest` is only used internally (this is actually the only location it's used?). Should we un-export it?

Reply:
Perhaps in a follow-up, as it's not been introduced in this PR? 😅